src/controller/CHIPCluster.cpp

```diff
@@ -42,6 +42,37 @@ CHIP_ERROR ClusterBase::Associate(DeviceProxy * device, EndpointId endpoint)
     return err;
 }
 
+CHIP_ERROR ClusterBase::Associate(GroupId groupId, DeviceProxy * device)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    // TODO: Check if the device supports mCluster at the requested endpoint
+
+    mDevice = device;
+    if (mDevice->GetSecureSession().HasValue())
+    {
+        // Local copy to preserve original SessionHandle for future Unicast communication.
+        SessionHandle session = mDevice->GetSecureSession().Value();
+        session.SetGroupId(groupId);
+        mSessionHandle.SetValue(session);
+
+        // Sanity check
+        if (!mSessionHandle.Value().IsGroupSession())
+        {
+            err = CHIP_ERROR_INCORRECT_STATE;
+        }
+    }
+    else
+    {
+        // something fishy is going on
```

Comment:
Well, we may not have a CASE session to the device...  but that should not be a problem for group comms.

Please add a TODO for this whole function, because it's going to need various changes to actually work right.

Reply:
Done